controllers/openstackmachine_controller.go

```diff
@@ -340,11 +380,12 @@ func (r *OpenStackMachineReconciler) getOrCreate(computeService *compute.Service
 	return instance, nil
 }
 
-func handleMachineError(logger logr.Logger, openstackMachine *infrav1.OpenStackMachine, reason capierrors.MachineStatusError, message error) {
-	openstackMachine.Status.FailureReason = &reason
+func handleUpdateMachineError(logger logr.Logger, openstackMachine *infrav1.OpenStackMachine, message error) {
+	err := capierrors.UpdateMachineError
+	openstackMachine.Status.FailureReason = &err
 	openstackMachine.Status.FailureMessage = pointer.StringPtr(message.Error())
 	// TODO remove if this error is logged redundantly
-	logger.Error(fmt.Errorf(string(reason)), message.Error())
+	logger.Error(fmt.Errorf(string(err)), message.Error())
```

Comment:
Any particular reason to `fmt.Errorf(string(err))` here instead of just using `err`?

Reply:
Oh that looks pretty useless :P

Reply:
Looks like capierrors.UpdateMachineError does not implement the error interface

Reply:
Ah, that makes sense then, thanks for clarifying.

Reply:
But it really looks strang :)